### DISCUSSION TITLE
Relationship between VALI, VPF, and PyNvVideoCodec

### DISCUSSION THREAD

**Discussion by tvercaut** [RomanArzumanyan/VALI#33] [Q&A] [answered]
Hi,

I am interested in using nvenc / nvdec from python and initially came across VPF. I now see that both VALI and PyNvVideoCodec pose themselves as successors of VPF:
https://catalog.ngc.nvidia.com/orgs/nvidia/resources/pynvvideocodec
https://github.com/RomanArzumanyan/VALI?tab=readme-ov-file#vali

How do the 2 options compare?

Best wishes,
Tom

**Comment by RomanArzumanyan** [comment, marked as answer]
Hi @tvercaut 

I’ve started VPF back at my Nvidia days. Then I left the company and had been supporting VPF for some time.

A while ago Nvidia decided to develop PyNvVideoCodec as a closed source project. I don’t know too much what are the reasons behind this decision.

Anyway, I didn’t want to leave VPF behind and started VPF spin off. With VPF repo admins permission I’m redirecting VPF issues towards this repo.

We maintain good relationship with NV folks, they were very nice to me and even choose a different module naming for their new project.

WRT PyNvVideoCodec feature set, how it compares vs VPF or VALI - honestly I don’t know anything about that.

**Reply by tvercaut** [reply]
Thanks. That's helpful.
